lib/Sema/CSFix.cpp

```diff
@@ -89,13 +89,21 @@ ForceDowncast *ForceDowncast::create(ConstraintSystem &cs, Type fromType,
 
 bool ForceOptional::diagnose(const Solution &solution, bool asNote) const {
   MissingOptionalUnwrapFailure failure(solution, getFromType(), getToType(),
-                                       getLocator());
+                                       UnwrapCount, getLocator());
```

Comment:
So this is where the fun begins, see my comment in the test case :) I think we need to "stack" the failures while diagnosing a force-unwrap fix because it requires `??` per level of optional unwrapped.